compiler/rustc_span/src/lib.rs

```diff
@@ -1016,22 +1019,43 @@ impl Default for Span {
     }
 }
 
-impl<E: Encoder> Encodable<E> for Span {
-    default fn encode(&self, s: &mut E) {
-        let span = self.data();
-        span.lo.encode(s);
-        span.hi.encode(s);
+pub trait SpanEncoder: Encoder {
```

Comment:
I think it would be nice if there were some doc comments here saying that this is used for why and it exists.

Reply:
Done